libc/signal.c

```diff
@@ -0,0 +1,196 @@
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	/* Since these function shall not return a value, we check only errno */
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], "Lorem Ipsum");
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, MAX_LEN_STRING, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, "Lorem Ipsum"));
+		fflush(err_file);
+	}
+	fclose(err_file);
+#endif
+}
+
+
+TEST(signal_psignal, ascii_string)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	char buff[MAX_LEN_STRING];
+
+	char ascii_str[2 * MAX_LEN_STRING];
+
+	for (int i = 0; i < 128; i++) {
+		ascii_str[i] = i;
+	}
+	ascii_str[128] = '\0';
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], ascii_str);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, MAX_LEN_STRING, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, ascii_str));
+		fflush(err_file);
+	}
+	fclose(err_file);
+#endif
+}
+
+
+TEST(signal_psignal, psignal_strsignal)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		int length;
+		char *signal_output = strsignal(signal_codes[i]);
+
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], "Simple Message");
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		length = strlen(signal_output) + strlen("Simple Message: ") + 1;
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, length, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, "Simple Message"));
+		TEST_ASSERT_NOT_NULL(strstr(buff, signal_output));
+		fflush(err_file);
+	}
+
+	fclose(err_file);
+#endif
+}
+
+
+TEST(signal_psignal, psignal_strsignal_null)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		char *signal_output = strsignal(signal_codes[i]);
+
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], NULL);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_EQUAL_STRING(signal_output, fgets(buff, strlen(signal_output) + 1, err_file));
```

Comment:
try to take more bytes or You would miss potential garbage data after the signal name